### ISSUE TITLE
potential regression in helm chart handling of `proxyListenerMode`

### ISSUE THREAD

**Issue by fspmarshall** [gravitational/teleport#24302]
When upgrading from v11 to v12 helm chart (`gcp` mode), a user had their agents lose connectivity. 

After some investigation, it became apparent that the agents were all setup under the assumption that the proxy listener mode was `multiplex` (i.e. TLS routing enabled), but examining the post-upgrade control plane, multiplexing was not enabled in the auth config file.  The user's helm values had `proxyListenerMode: ""`.  Switching this to `multiplex` and redeploying fixed their connectivity issues.

We were unable to examine the actual previous configuration, but the relevant files were all under version control, so the customer is reasonably certain that no configuration was changed on their end.  We should therefore investigate the possibility of a regression in helm chart behavior when proxy listener mode is not specified.

**Comment by webvictim** [comment]
## Differences

Some changes I've noticed between v11 and v12:

`branch/v11`:
- `auth_service.proxy_listener_mode` is **not set** by default, or set to `multiplex` if `proxyListenerMode: multiplex` set in values
- config version is **not set** by default, unless `proxyListenerMode` is set to `multiplex`, in which case it's `version: v2`
- proxy listen addresses are **not set** - only ` kube_listen_addr: 0.0.0.0:3026` is set

`branch/v12`:
- `auth_service.proxy_listener_mode` is **always** set; to `separate` by default, or `multiplex` if `proxyListenerMode: multiplex` is set in values
- config version is **always** `version: v3`
- when `proxyListenerMode: separate`, the following proxy listen addresses are **explicitly set**:
```yaml
proxy_service:
  listen_addr: 0.0.0.0:3023
  tunnel_listen_addr: 0.0.0.0:3024
  kube_listen_addr: 0.0.0.0:3026
  mysql_listen_addr: 0.0.0.0:3036
```

## Theory/configs

This probably comes down to a difference between how the following two configs are handled:

`branch/v11` default proxy config:
```yaml
proxy_service:
  enabled: true
  kube_listen_addr: 0.0.0.0:3026
```

This defaults to `version: v1` and should enable separate listeners by default as part of that.

`branch/v12` default proxy config:
```yaml
version: v3
proxy_service:
  enabled: true
  listen_addr: 0.0.0.0:3023
  tunnel_listen_addr: 0.0.0.0:3024
  kube_listen_addr: 0.0.0.0:3026
  mysql_listen_addr: 0.0.0.0:3036
```

It seems that setting a newer config version by default and manually specifying the listener addresses may result in slightly different behaviour to the previous v11 default.

**Comment by hugoShaka** [comment]
I agree with gus, I think this boils down to both:
- the v11 chart operating in both separate and multiplex modes by default (listeners are always defined)
- the config v1 to v3 change (disables listeners when not explicitly enabled)

If we look at [the TLS routing migration guide](https://goteleport.com/docs/management/operations/tls-routing/) we were operating by default at step 6/7.

I don't think the new behaviour is an issue per-se. It is aligned with the new Teleport default config. We can't really rollback to v1 because we use v3-sepcific features. I really messed up by not noticing and documenting the breaking change.

Another thing to consider is that even if we keep the current behaviour, we must provide a path to upgrade from non-multiplexing to multiplexing. Manually setting the listeners while running in TLS multiplexing mode might allow us to support both modes, but I think the service will not send traffic. AFAIK this issue was already in the v11 chart.

**Comment by hugoShaka** [comment]
I have a theory on what happened:

The user never specified `multiplex,` so their agents connected through separate listeners. (agents are basically brute-forcing all connection methods and will find a way as long as you point them to a proxy).

In v11 the only possible user value for `proxyListenerMode` was `multiplex`. We defaulted to an empty string.

In v12, we gave `proxyListsenerMode` two values: `separate` and `multiplex`. We changed the default to `separate`.

The user copied the full v11 values file, including all v11 defaults and used it. By doing this, they blocked us from changing the default values. When feeding the v12 chart with v11 values they set `proxyListenerMode` to `""` which is not supported by the v12 chart and disabled `separate` listeners while not activating TLS multiplexing.

Copying all default values is a harmful but common practice in the Helm ecosystem. I don't know what to think about this, is it our responsibility to provide backward compatibility for values that users must not set?

Anyway, I'll send a PR to handle the case where the user sets an invalid value for `proxyListenerMode`.
